src/app/utility/matchSearch.tsx

```diff
@@ -4,7 +4,7 @@ export function matchSearch(doctor: Doctor, searchValue: string): boolean {
   const concatedString =
     `${doctor.name}${doctor.surename}${doctor.adress}${doctor.city}`.toLowerCase();
   const searchValueLowerCase = searchValue.toLowerCase();
-  const regExpObj = new RegExp(`${searchValueLowerCase}*`, 'g');
+  const regExpObj = new RegExp(`${searchValueLowerCase}`, 'g');
   const result = concatedString.match(regExpObj);
   return result ? true : false;
```

Comment:
Your code is fine, just an idea to simplify it (because I like this codewars style challenge ;) )

```suggestion
  const doctor = [`${doctor.name doctor.surename}`, doctor.adress, doctor.city]
  const searchRegEx = new RegExp(searchValue, 'i');
  return doctor.some(item => searchRegEx.test(item))
```

In particular, 
- you can use the 'i' option in RegExp for case-insensitive search
- you can use `.test` instead of `.match` to get a boolean return directly
- I used an array so that I don't search for string combinations of e.g. name and address